liboptv/src/multimed.c

```diff
@@ -0,0 +1,526 @@
+      for (j = 0; j < 2; j++) {
+          pixel_to_metric (&x, &y, xc[i], yc[j], cpar);
+
+          x = x - cal->int_par.xh;
+          y = y - cal->int_par.yh;
+  
+          correct_brown_affin (x, y, cal->added_par, &x,&y);  
+          ray_tracing(x,y, cal, *(cpar->mm), pos, a);
+          
+          move_along_ray(Zmin, pos, a, xyz);
+          trans_Cam_Point(cal->ext_par, *(cpar->mm), cal->glass_par, xyz, \
+            &Ex_t, xyz_t, (double *)cross_p, (double *)cross_c);
+
+          if( xyz_t[2] < Zmin_t ) Zmin_t = xyz_t[2];
+          if( xyz_t[2] > Zmax_t ) Zmax_t = xyz_t[2];
+
+          R = norm((xyz_t[0] - Ex_t.x0), (xyz_t[1] - Ex_t.y0), 0);
+          if (R > Rmax)
+              Rmax = R;
+           
+          move_along_ray(Zmax, pos, a, xyz);
+          trans_Cam_Point(cal->ext_par, *(cpar->mm), cal->glass_par, xyz,\
+              &Ex_t, xyz_t, (double *)cross_p, (double *)cross_c);
+  
+          if( xyz_t[2] < Zmin_t ) Zmin_t = xyz_t[2];
+          if( xyz_t[2] > Zmax_t ) Zmax_t = xyz_t[2];
+
+          R = norm((xyz_t[0] - Ex_t.x0), (xyz_t[1] - Ex_t.y0), 0);
+          if (R > Rmax)
+              Rmax = R;
+      }
+  }
+
+  /* round values (-> enlarge) */
+  Rmax += (rw - fmod (Rmax, rw));
+
+  /* get # of rasterlines in r,z */
+  nr = (int)(Rmax/rw + 1);
+  nz = (int)((Zmax_t-Zmin_t)/rw + 1);
+
+  /* create two dimensional mmlut structure */
+  vec_set(cal->mmlut.origin, Ex_t.x0, Ex_t.y0, Zmin_t);
+  
+  cal->mmlut.nr = nr;
+  cal->mmlut.nz = nz;
+  cal->mmlut.rw = rw;
+  
+  // if (cal.mmlut.data != NULL)			// preventing memory leaks, ad holten, 04-2013
+  //	free (cal.mmlut.data);
+  if (cpar->mm->lut == 0){
+      cal->mmlut.data = (double *) malloc (nr*nz * sizeof (double));
+  
+      /* fill mmlut structure */
+      Ri = (double *) malloc (nr * sizeof (double));
+      for (i = 0; i < nr; i++)
+        Ri[i] = i*rw;
+
+      Zi = (double *) malloc (nz * sizeof (double));
+      for (i = 0; i < nz; i++)
+        Zi[i] = Zmin_t + i*rw;
+
+      for (i = 0; i < nr; i++) {
+        for (j = 0; j < nz; j++) {
+            xyz[0] = Ri[i] + Ex_t.x0;
+            xyz[1] = Ex_t.y0;
+            xyz[2] = Zi[j];
+    
+            cal->mmlut.data[i*nz + j] = multimed_r_nlay(cal, cpar->mm, xyz);
+        } /* nr */
+      } /* nz */
+    
+      free (Ri);	// preventing memory leaks, Ad Holten, 04-2013
+      free (Zi);
+
+      /* when finished initalization, change the setting of the LUT flag */
+      cpar->mm->lut = 1;
+    }
+}
+
+
+/*  get_mmf_from_mmlut() returns the value of mmf (double) for a 3D point
+    using the multimedia look up table.
+    
+    Arguments:
+    double position in 3D space, X,Y,Z
+    multimedia look-up table mmlut
+*/
+double get_mmf_from_mmlut (Calibration *cal, vec3d pos){
+  int i, ir,iz, nr,nz, rw, v4[4];
+  double R, sr, sz, mmf = 1.0;
+  double X,Y,Z;
+  
+  X = pos[0]; Y = pos[1]; Z = pos[2];
+  rw =  cal->mmlut.rw;
+  
+  if (X == 1.0 && Y == 1.0 && Z == 1.0){

```

Comment:
The bodz of the ``if`` is the same as the body of the ``else``, or am I 
getting crazy?

On 02/10/15 15:07, Alex Liberzon wrote:
> In liboptv/src/multimed.c
> <https://github.com/OpenPTV/openptv/pull/60#discussion_r41014687>:
>
>> +/*  get_mmf_from_mmlut() returns the value of mmf (double) for a 3D point
>> +    using the multimedia look up table.
>> +
>> +    Arguments:
>> +    double position in 3D space, X,Y,Z
>> +    multimedia look-up table mmlut
>> +*/
>> +double get_mmf_from_mmlut (Calibration *cal, vec3d pos){
>> +  int i, ir,iz, nr,nz, rw, v4[4];
>> +  double R, sr, sz, mmf = 1.0;
>> +  double X,Y,Z;
>> +
>> +  X = pos[0]; Y = pos[1]; Z = pos[2];
>> +  rw =  cal->mmlut.rw;
>> +
>> +  if (X == 1.0 && Y == 1.0 && Z == 1.0){
>
> i do not understand this comment
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenPTV/openptv/pull/60/files#r41014687>.
>